src/utils/scoring/BestElementFinder.cpp

```diff
@@ -47,17 +49,32 @@ std::optional<size_t> cura::BestElementFinder::findBestElement(const size_t cand
                 iterator->score += weighed_criterion.criterion->computeScore(iterator->candidate_index) * weighed_criterion.weight;
             }
 
-            best_score = std::max(best_score, iterator->score);
+            if (best_candidate_iterator == end || iterator->score > best_candidate_iterator->score)
```

Comment:
We could also set the initial score(s) to `numeric_limits<...>::min()` (if the type is integer, or negative infinity if the type is float). That way, we don't need the end iterator. -- But this works too.

Reply:
Well I need this iterator anyway because in the early-out I need the index. And I also stopped used min and max as markers for a specific state, I think the previous version used an std::optional